resources/snakefiles/Snakefile.smk

```diff
@@ -0,0 +1,242 @@
+          -fq2 {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq \
+          2>> {log.unmapped}
+
+        # Convert sorted mapped alignment to fastq format
+        bedtools bamtofastq -i {output.temp_dir}/{wildcards.sample}.mapped.bam \
+          -fq {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq \
+          -fq2 {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq \
+          2>> {log.mapped}
+
+        # zip the filtered fastqs
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq.gz
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq.gz
+
+        # zip the host fastqs
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq.gz
+        pigz -p {threads} \
+          -c {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq > \
+          {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq.gz
+
+        # copy the filtered fastqs to final location
+        cp {output.temp_dir}/{wildcards.sample}.R1.trimmed.filtered.fastq.gz \
+          {output.nonhost_R1}
+        cp {output.temp_dir}/{wildcards.sample}.R2.trimmed.filtered.fastq.gz \
+          {output.nonhost_R2}
+
+        # copy the host fastqs to final location
+        cp {output.temp_dir}/{wildcards.sample}.R1.trimmed.host.fastq.gz \
+          {output.host_R1}
+        cp {output.temp_dir}/{wildcards.sample}.R2.trimmed.host.fastq.gz \
+          {output.host_R2}
+        """
+
+
+rule metaspades_assembly:
+    """
+
+    Performs a metagenomic assembly on a sample using MetaSPAdes.
+
+    """
+    input:
+        fastq1=rules.host_filter.output.nonhost_R1,
+        fastq2=rules.host_filter.output.nonhost_R2
+    output:
+        contigs="output/metaspades/{sample}.contigs.fasta",
+        temp_dir=temp(directory("output/{sample}_temp"))
+    conda:
+        "resources/envs/spades.yaml"
+    threads:
+        config['threads']['assembly']
+    benchmark:
+        "output/benchmarks/metaspades/{sample}_benchmark.txt"
+    shell:
+        """
+        # Make temporary output directory
+        mkdir -p {output.temp_dir}
+
+        # Make directory for assemblies and log files
+        mkdir -p output/metaspades/
+        mkdir -p output/logs/metaspades/
+
+        # run the metaspades assmebly
+        metaspades.py --threads {threads} \
+          -o output/{output.temp_dir}/ \
+          --pe1-1 {input.fastq1} \
+          --pe1-2 {input.fastq2}
+
+        # move and rename the contigs file into a permanent directory
+        mv output/{output.temp_dir}/contigs.fasta output/metaspades/{wildcards.sample}.contigs.fasta
+
+        # move the log file
```

Comment:
A better way to do logging is use the in-built Snakemake logfile funciton. In this case, you can define a log file as one of the top-level keywords in the rule. (See earlier rules for examples.) Then you can pipe the stderr from a command to that log file. 

```{bash}
example.sh > stdout.txt 2> stderr.log 
```
 The `2>` is directing stderr (2) to a log file, while stdout (implicit 1) is going to an output text file. So in the snakemake case, it looks like:

```{bash}
example.sh > {output} 2> {log}
```

Reply:
Gotcha. The log file was already being written to the output directory, so I figured it was just as easy to move it. I'll change this though.

Reply:
I made this change in the latest commit, but the log files do not appear to have anything in them. Can you see anything wrong with them?

Reply:
Hmm, it's possible that spades is dumping log to stdout and not stderr. You can redirect stdout to stderr (and hence capture in log) with:

`2> {log} 1>&2`

Reply:
This fixed it, thanks!